project/src/main/java/com/google/sps/Servlets/SharingServlet.java

```diff
@@ -0,0 +1,35 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.*;
+import com.google.sps.models.Database;
+import com.google.sps.models.User;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+@WebServlet("/Share")
+@MultipartConfig
+public class SharingServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String email = request.getParameter("email");
+    String hash = request.getParameter("documentHash");
+    PrintWriter out = response.getWriter();
+
+    System.out.println(email + hash);
+
+    if(Database.shareDocument(hash, email)) {
+      out.println("Document shared with " + email);
```

Comment:
does this servlet redirect back to the original document this user was on? might be useful to pass this info back to that page

Reply:
Since I am making a call with AJAX, the page is never reloaded. This text is being returned to the function where the call was originally made, and then being displayed on the page